### PR TITLE
add capability to set offset and read ID registers

### PR DESCRIPTION
**Pull request by amitrao17**
Hi,

I'm using the VL6180X and needed to do two things which were not implemented in the library:

1 - change the offset register to 'calibrate' the range
2 - read the ID bytes

These have been implemented and I've tested them to be functional.

I've never done a pull request so be gentle!

Thanks

### CODE REVIEW

File: Adafruit_VL6180X.cpp [hunk 1]
```diff
@@ -396,6 +396,44 @@ float Adafruit_VL6180X::readLux(uint8_t gain) {
...
+  id[0] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 0);
+  id[1] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 1);
+  id[2] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 2);
+  id[3] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 3);
+  id[4] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 4);
+  id[5] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 5);
+  id[6] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 6);
+  id[7] = read8(VL6180X_REG_IDENTIFICATION_MODEL_ID + 7);
+
+  id_ptr = &id[0];
```
Comment (ladyada) [inline Adafruit_VL6180X.cpp]: this doesn't work, the `id` variable will disappear when its returned. you should assign into `id-ptr` and require the caller to have 8 bytes available

**Review by ladyada — CHANGES_REQUESTED** [review]

File: Adafruit_VL6180X.cpp (hunk 1 above)
Comment (caternuson) [reply Adafruit_VL6180X.cpp]: Also check offsets. Looks like there is nothing at `+ 5`?
image

**Comment by caternuson** [comment]
See here for info on clang formatting to help pass the CI checks:
https://learn.adafruit.com/the-well-automated-arduino-library/formatting-with-clang-format

**Review by ladyada — APPROVED** [review]

File: Adafruit_VL6180X.cpp (hunk 1 above)
Comment (amitrao17) [reply Adafruit_VL6180X.cpp]: thanks @ladyada and @caternuson good points

**Review by ladyada — APPROVED** [review]

**Comment by amitrao17** [comment]
I could update the example to show the new functions being used if that is helpful?

**Pull request merged**